src/lib_json/json_reader.cpp

```diff
@@ -1378,30 +1378,39 @@ bool OurReader::readCppStyleComment() {
   }
   return true;
 }
-
-bool OurReader::readNumber(bool checkInf) {
+// if checkNeg is true, read negative number
+bool OurReader::readNumber(bool checkNeg) {
   const char* p = current_;
-  if (checkInf && p != end_ && *p == 'I') {
-    current_ = ++p;
-    return false;
+  char c = '0'; // stopgap for already consumed character  
+  if(checkNeg){
+    if (checkNeg && p != end_ && *p == 'I') {
+      current_ = ++p;
+      return false;
+    }
+    else
+    {
+      current_ = ++p;                // skip '-'
+      OurReader::readNumber(false);  // read the back part of the '-'
```

Comment:
Nit: I'd prefer an early return here instead of wrapping all of the parsing code below in an else.

Reply:
I think I should explain a little bit why I modified it this way.
The original method **readNumber(bool checkInf)** looks like it can read number(positive or negtive) and check if it's `negtive Infinity` or not.  And in fact it read number positive or negtive that depend on a flag `checkInf`, this seems strange, right? 
Moreover, when `checkInf` is **true**, that's because it's in  `-` case, then, when `readNumber(true)` return flase, we think it's `-Infinity`, all other cases it return true, and we  think it's a negtive number, and set token's type to `tokenNumber`, But this is where the problem is. when it's in `-` case, and it's actually like `-i` or `->-->--><!--#set var=`  or something else that begin with  `-`, we still take it as a number, and token's type to `tokenNumber`. 
So I modify the method to **readNumber(bool checkNeg)** it looks more clear and unambiguous. it can read positive number or negtive number(which include negtive Infinity) by the flag **checkNeg**, and more important, it can selove the problem as I mentioned above. And the relative testcase I add has passed in my PC, so I think it really works.